network/horizon/admin-guide/error-handling.mdx

```diff
@@ -0,0 +1,182 @@
+
+There are some resolution strategies that are common between the 2 endpoints while some strategies are more endpoint specific.
+
+### Request Adjustments
+
+Certain transaction submission failures also need adjustments to succeed.
+
+- If the XDR is malformed, or the transaction is otherwise invalid, you’ll encounter a `400 Bad Request` (for example, see excluding a source account). Both transactions and their operations can be easily malformed: look at the `extras.result_codes` field for details and cross-reference them with the appropriate result codes documentation to determine specifics.
+- Transaction fees are also a safe adjustment. If you get a `tx_insufficient_fee` error, refer to the Insufficient Fees and Surge Pricing section later in this document.
+
+### Polling and Retrying Transactions
+
+#### Async Transaction Submission
+
+Submissions using the async endpoint return a clear response back from core. There are different actions that clients can take based on the specific `tx_status` returned:
+
+1. `PENDING`: the submission is successful but the txn is still waiting to be included in the ledger. You should use a `GET` endpoint like `/transactions` to poll for the submitted txn and check if it makes into the ledger.
+2. `ERROR`: the submission did not go through due to some error with core. Take a look at the attached error message and retry the submission again.
+3. `TRY_AGAIN_LATER`: this indicates that the stellar-core instance is currently unable to process any submission requests. Clients should wait for sometime before retrying it again.
+
+#### Non-Async Transaction Submission
+
+Due to the blocking nature of the non-async endpoint, things are a little different compared to the async strategy. There are 3 possible scenarios that clients can encounter:
+
+1. The submission is successful and Horizon returns the transaction response back. This is the happy path and clients do not need to do anything but wait for Horizon's response.
+2. Core sends back an `ERROR` response from the submission. Clients should consult the attached error message and retry the submission again.
+3. Timeouts: Horizon may send a `504 Timeout` after transaction submission. Timeouts are not errors but warnings that your request hasn’t been fulfilled yet. This can happen because of the relationship between Horizon and Stellar Core- the network may take some time (5-10 mins during congestion) to accept the transaction. At the same time, Horizon needs to provide developers with a response within 30 seconds. Due to the blocking nature of the endpoint, there could be many possible scenarios behind the timeout - transient outages, congestion on the network, issues on Core side, issues on Horizon side etc.... There is no way to cancel a transaction after submitting it. So, you need to carefully consider before resubmitting your transactions.
```

Comment:
```suggestion
3. Timeouts: Horizon may respond with a `504` HTTP code. This response is not an error but a warning that your transaction hasn't been accepted by the network yet. There could be many possible scenarios behind the timeout, the most common of which is network congestion, but it could also be due to other transient issues. There are a couple things you should to do best handle this response.
```